utilities/install.py

```diff
@@ -1,15 +1,78 @@
-import site, sys, importlib
+import importlib
+import logging
+import os
+import site
+import subprocess
+import sys
 
+from .. import global_data
+
+logger = logging.getLogger(__name__)
 
-def check_module(package):
-    # Note: Blender might be installed in a directory that needs admin rights and thus defaulting to a user installation.
-    # That path however might not be in sys.path....
 
+def check_module(package):
+    refresh_path()
+    # This was done in the past to make sure modules in usersite can be found.
+    # Since using USERBASE this shouldnt be necessary, but it's left as a fallback.
     p = site.USER_SITE
     if p not in sys.path:
-        sys.path.append(p)
+        sys.path.insert(0, p)
     try:
         importlib.import_module(package)
 
     except ModuleNotFoundError as e:
         raise e
+
+
+def install_pip() -> int:
+    import ensurepip
+
+    # ensurepip.bootstrap just hides the return
+    # code which is needed for the currect error messages.
+    ret_val = ensurepip._bootstrap()
+    # Env var left over from pip installation, needs to be removed
+    os.environ.pop("PIP_REQ_TRACKER", None)
+    return ret_val
+
+
+def check_pip_installed() -> bool:
+    cmd = [global_data.PYPATH, "-m", "pip", "--version"]
+    if subprocess.call(cmd, env=global_data.env_vars) == 0:
+        return True
+    return False
```

Comment:
Also this doesn't seem to function. In the terminal stdout/err I get `No module named pip` when the return value is `0`. Which is very strange. In turn, it doesn't want to install pip.

I do run on Linux. Given the difference for our platforms, I will try to investigate myself/

Reply:
As a fix, I would recommend to always use `ensurepip` as it will, at most, update pip to the latest version in case pip is already present.

> This invocation will install `pip` if it is not already installed, but otherwise does nothing.